admin-dev/themes/new-theme/js/notifications.js

```diff
@@ -60,35 +62,50 @@ const refreshNotifications = function () {
   clearTimeout(timer);
 }
 
+let replaceGlobally = function (original, searchTxt, replaceTxt) {
+  const regex = new RegExp(searchTxt, 'g');
+  return original.replace(regex, replaceTxt) ;
+}
+
 let fillTpl = function (results, eltAppendTo, tpl) {
   eltAppendTo.children('.notification-elements').empty();
   if (results.length > 0) {
+    let router = new Router();
     eltAppendTo.removeClass('empty');
     $.each(results, function (property, value) {
       if (undefined === tpl) {
         return;
       }
 
-      eltAppendTo.children('.notification-elements').append(
-        tpl.replace(/_id_order_/g, parseInt(value.id_order))
-          .replace(/_customer_name_/g, value.customer_name)
-          .replace(/_iso_code_/g, value.iso_code)
-          .replace(/_carrier_/g, (value.carrier !== "" ? ` - ${value.carrier}` : ""))
-          .replace(/_total_paid_/g, value.total_paid)
-          .replace(/_id_customer_/g, parseInt(value.id_customer))
-          .replace(/_company_/g, (value.company !== "" ? ` (${value.company}) ` : ""))
-          .replace(/_date_add_/g, value.date_add)
-          .replace(/_status_/g, value.status)
-          .replace(/order_url/g, `${baseAdminDir}index.php?tab=AdminOrders&token=${token_admin_orders}&vieworder&id_order=${value.id_order}`)
-          .replace(/customer_url/g, `${baseAdminDir}index.php?tab=AdminCustomers&token=${token_admin_customers}&viewcustomer&id_customer=${value.id_customer}`)
-          .replace(/message_url/g, `${baseAdminDir}index.php?tab=AdminCustomerThreads&token=${token_admin_customer_threads}&viewcustomer_thread&id_customer_thread=${value.id_customer_thread}`)
-      );
+      let tplReplaced = tpl.replace(/_id_order_/g, parseInt(value.id_order))
+        .replace(/_customer_name_/g, value.customer_name)
+        .replace(/_iso_code_/g, value.iso_code)
+        .replace(/_carrier_/g, (value.carrier !== "" ? ` - ${value.carrier}` : ""))
+        .replace(/_total_paid_/g, value.total_paid)
+        .replace(/_id_customer_/g, parseInt(value.id_customer))
+        .replace(/_company_/g, (value.company !== "" ? ` (${value.company}) ` : ""))
+        .replace(/_date_add_/g, value.date_add)
+        .replace(/_status_/g, value.status)
+        .replace(/order_url/g, `${baseAdminDir}index.php?tab=AdminOrders&token=${token_admin_orders}&vieworder&id_order=${value.id_order}`)
+
+      let customerThreadId = parseInt(value.id_customer_thread);
+      if (customerThreadId) {
+        let route = router.generate('admin_customer_threads_view', {customerThreadId});
+        tplReplaced = replaceGlobally(tplReplaced, 'message_url', route);
+      }
+
+      let customerId = parseInt(value.id_customer);
+      if (customerId) {
+        let route = router.generate('admin_customers_view', {customerId});
```

Comment:
You should declare `route` and `tplReplaced` once

Reply:
declared outside foreach, hope it is what you expected. If not, let me know